services/painting-process-equipment-defect-detection-model-service/app/dependencies.py

```
@@ -0,0 +1,81 @@
+        print("모델 로드 완료")
+
+        # 모델 로드 성공 후 Explainer 재생성
+        _explainer = shap.TreeExplainer(_model)
+        print("SHAP explainer 재생성 완료")
+
+        _initialization_complete = True # 모든 리소스 로드 성공 시 플래그를 True로 설정
+        print("애플리케이션 초기화 완료")
+
+    except FileNotFoundError:
+        print("오류: 모델 파일을 찾을 수 없습니다. 경로를 확인해주세요.")
+        raise FileNotFoundError("Model file not found at specified path.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)